### DISCUSSION TITLE
Best way to handle giving credits to user on start of each subscription billing cycle?

### DISCUSSION THREAD

**Discussion by alhafoudh** [pay-rails/pay#763] [Q&A] [answered]
What would be the best way to handle giving credits to user on start of each subscription billing cycle? I need to "top up" credits each month/year.

I already have a credit tracking system in place using plutus gem.

My first idea is to schedule a job every day to check for subscriptions which have billing cycle starting that day and give the credits.

**Comment by excid3** [comment, marked as answer]
I would use webhooks to add credits when the subscription is charged.
